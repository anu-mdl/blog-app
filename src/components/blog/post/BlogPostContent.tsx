'use client';

import { useEffect } from 'react';

interface BlogPostContentProps {
  content: string;
}

export function BlogPostContent({ content }: BlogPostContentProps) {
  useEffect(() => {
    const headings = document.querySelectorAll('h1, h2, h3, h4, h5, h6');
    headings.forEach(heading => {
      if (!heading.id) {
        const id = heading.textContent
          ?.toLowerCase()
          .replace(/[^a-z0-9]+/g, '-')
          .replace(/(^-|-$)/g, '');
        if (id) {
          heading.id = id;
        }
      }
    });
  }, [content]);

  const formatContent = (text: string) => {
    return text.split('\n').map((line, index) => {
      if (line.startsWith('# ')) {
        return (
          <h1 key={index} className="text-3xl font-bold mt-8 mb-4 first:mt-0">
            {line.slice(2)}
          </h1>
        );
      }
      if (line.startsWith('## ')) {
        return (
          <h2 key={index} className="text-2xl font-bold mt-8 mb-4">
            {line.slice(3)}
          </h2>
        );
      }
      if (line.startsWith('### ')) {
        return (
          <h3 key={index} className="text-xl font-bold mt-6 mb-3">
            {line.slice(4)}
          </h3>
        );
      }

      if (line.trim() === '') {
        return <br key={index} />;
      }

      return (
        <p key={index} className="mb-4 leading-relaxed text-foreground/90">
          {line}
        </p>
      );
    });
  };

  return (
    <div className="prose prose-lg max-w-none prose-headings:text-foreground prose-p:text-foreground/90 prose-strong:text-foreground prose-code:text-foreground prose-pre:bg-muted prose-pre:text-foreground">
      {formatContent(content)}
    </div>
  );
}
