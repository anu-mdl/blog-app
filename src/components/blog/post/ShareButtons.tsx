'use client';

import { Button } from '@/components/ui/button';
import { Facebook, Twitter, Linkedin, Link, Copy } from 'lucide-react';
import { useState } from 'react';

interface ShareButtonsProps {
  url: string;
  title: string;
}

export function ShareButtons({ url, title }: ShareButtonsProps) {
  const [copied, setCopied] = useState(false);

  const handleCopyLink = async () => {
    try {
      await navigator.clipboard.writeText(url);
      setCopied(true);
      setTimeout(() => setCopied(false), 2000);
    } catch (err) {
      console.error('Failed to copy link:', err);
    }
  };

  const shareUrls = {
    twitter: `https://twitter.com/intent/tweet?text=${encodeURIComponent(title)}&url=${encodeURIComponent(url)}`,
    facebook: `https://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(url)}`,
    linkedin: `https://www.linkedin.com/sharing/share-offsite/?url=${encodeURIComponent(url)}`
  };

  return (
    <div className="flex items-center gap-2">
      <span className="text-sm font-medium text-muted-foreground mr-2">
        Share:
      </span>

      <Button variant="outline" size="icon" asChild>
        <a href={shareUrls.twitter} target="_blank" rel="noopener noreferrer">
          <Twitter className="h-4 w-4" />
          <span className="sr-only">Share on Twitter</span>
        </a>
      </Button>

      <Button variant="outline" size="icon" asChild>
        <a href={shareUrls.facebook} target="_blank" rel="noopener noreferrer">
          <Facebook className="h-4 w-4" />
          <span className="sr-only">Share on Facebook</span>
        </a>
      </Button>

      <Button variant="outline" size="icon" asChild>
        <a href={shareUrls.linkedin} target="_blank" rel="noopener noreferrer">
          <Linkedin className="h-4 w-4" />
          <span className="sr-only">Share on LinkedIn</span>
        </a>
      </Button>

      <Button variant="outline" size="icon" onClick={handleCopyLink}>
        {copied ? (
          <Copy className="h-4 w-4 text-green-600" />
        ) : (
          <Link className="h-4 w-4" />
        )}
        <span className="sr-only">{copied ? 'Link copied' : 'Copy link'}</span>
      </Button>
    </div>
  );
}
