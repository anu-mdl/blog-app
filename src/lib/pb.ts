import PocketBase from "pocketbase";

const POCKETBASE_URL = process.env.NEXT_PUBLIC_POCKETBASE_URL || "http://127.0.0.1:8090";

export const pb = new PocketBase(POCKETBASE_URL);

// (опционально) настраиваем автологин из cookie при SSR/SSG
if (typeof window !== "undefined" && pb.authStore.isValid) {
  pb.authStore.loadFromCookie(document.cookie);
}