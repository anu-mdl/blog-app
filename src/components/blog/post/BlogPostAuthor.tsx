import { UsersRecord } from '@/api/api_types';
import { UserSocials } from '@/api/extended_types';
import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar';
import { Button } from '@/components/ui/button';
import { Card, CardContent } from '@/components/ui/card';
import { pb } from '@/lib/pb';
import { Github, Linkedin, Twitter } from 'lucide-react';

export function BlogPostAuthor({
  author
}: {
  author: UsersRecord<UserSocials>;
}) {
  return (
    <Card>
      <CardContent className="p-6">
        <div className="flex flex-col sm:flex-row gap-4">
          <Avatar className="h-16 w-16">
            <AvatarImage
              src={
                author?.avatar
                  ? pb.files.getURL(author, author.avatar)
                  : '/image.svg'
              }
              alt={author.username}
            />
            <AvatarFallback className="text-lg">
              {author.username?.charAt(0)}
            </AvatarFallback>
          </Avatar>

          <div className="flex-1 space-y-3">
            <div>
              <h3 className="text-lg font-semibold">About {author.username}</h3>
              <p className="text-muted-foreground mt-1">{author.bio}</p>
            </div>

            <div className="flex gap-2">
              {author.socials?.twitter && (
                <Button variant="outline" size="icon" asChild>
                  <a
                    href={`https://twitter.com/${author.socials.twitter}`}
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    <Twitter className="h-4 w-4" />
                    <span className="sr-only">Twitter</span>
                  </a>
                </Button>
              )}
              {author.socials?.github && (
                <Button variant="outline" size="icon" asChild>
                  <a
                    href={`https://github.com/${author.socials.github}`}
                    target="_blank"
                    rel="noopener noreferrer"
                  >
                    <Github className="h-4 w-4" />
                    <span className="sr-only">GitHub</span>
                  </a>
                </Button>
              )}
            </div>
          </div>
        </div>
      </CardContent>
    </Card>
  );
}
