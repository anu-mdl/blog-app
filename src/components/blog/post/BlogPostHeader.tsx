import { Badge } from '@/components/ui/badge';
import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar';
import { Clock, Calendar, MessageSquare } from 'lucide-react';
import { formatDate } from '@/lib/utils';
import { PostsRecordExtended } from '@/api/extended_types';
import { pb } from '@/lib/pb';
import Image from 'next/image';

export function BlogPostHeader({ post }: { post: PostsRecordExtended }) {
  const commentCount = 6;
  return (
    <header className="space-y-6">
      <div className="space-y-4">
        <Badge variant="secondary" className="w-fit">
          {post.category}
        </Badge>

        <h1 className="text-3xl md:text-4xl lg:text-5xl font-bold tracking-tight leading-tight">
          {post.title}
        </h1>

        <p className="text-xl text-muted-foreground leading-relaxed">
          {post.excerpt}
        </p>
      </div>

      <div className="flex flex-col sm:flex-row sm:items-center gap-4 text-sm text-muted-foreground">
        <div className="flex items-center gap-3">
          <Avatar className="h-10 w-10">
            <AvatarImage
              src={
                post.expand?.author?.avatar
                  ? pb.files.getUrl(
                      post.expand.author,
                      post.expand.author.avatar
                    )
                  : '/image.svg'
              }
              alt={post.expand?.author?.username}
            />
            <AvatarFallback>
              {post.expand?.author?.username?.charAt(0) || '?'}
            </AvatarFallback>
          </Avatar>
          <div>
            <div className="font-medium text-foreground">
              {post.expand?.author?.username}
            </div>
          </div>
        </div>

        <div className="flex items-center gap-4">
          <div className="flex items-center gap-1">
            <Calendar className="h-4 w-4" />
            <span>{formatDate(post.created || '')}</span>
          </div>
          <div className="flex items-center gap-1">
            <MessageSquare className="h-4 w-4" />
            <span>{commentCount} comments</span>
          </div>
        </div>
      </div>

      <div className="relative aspect-video overflow-hidden rounded-lg">
        <Image
          src={post.image ? pb.files.getURL(post, post.image) : '/image.svg'}
          alt="Cover"
          className="w-full h-full object-cover"
        />
      </div>

      <div className="flex flex-wrap gap-2">
        {post.tags?.split(',').map(tag => (
          <Badge key={tag} variant="outline" className="text-xs">
            #{tag.toLowerCase().replace(/\s+/g, '')}
          </Badge>
        ))}
      </div>
    </header>
  );
}
