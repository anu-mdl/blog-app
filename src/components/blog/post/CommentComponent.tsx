import { CommentsRecordExtended } from '@/api/extended_types';
import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar';
import { formatDate } from '@/lib/utils';
import { Heart } from 'lucide-react';

export const CommentComponent = ({
  comment,
  isReply = false
}: {
  comment: CommentsRecordExtended;
  isReply?: boolean;
}) => (
  <div className={`space-y-3 ${isReply ? 'ml-12' : ''}`}>
    <div className="flex gap-3">
      <Avatar className="h-8 w-8 flex-shrink-0">
        <AvatarImage
          src={comment.author.avatar || '/image.svg'}
          alt={comment.author.username}
        />
        <AvatarFallback>{comment.author.username?.charAt(0)}</AvatarFallback>
      </Avatar>

      <div className="flex-1 min-w-0 space-y-2">
        <div className="flex items-center gap-2 text-sm">
          <span className="font-medium truncate">
            {comment.author.username}
          </span>
          <span className="text-muted-foreground whitespace-nowrap">
            {formatDate(comment.created || '')}
          </span>
        </div>

        <p className="text-sm leading-relaxed break-words hyphens-auto whitespace-pre-wrap word-break-break-word overflow-wrap-anywhere">
          {comment.content}
        </p>

        <div className="flex items-center gap-4 text-xs text-muted-foreground">
          <button className="flex items-center gap-1 hover:text-foreground transition-colors">
            <Heart className="h-3 w-3" />
            <span>{comment.likes}</span>
          </button>
        </div>
      </div>
    </div>
  </div>
);
