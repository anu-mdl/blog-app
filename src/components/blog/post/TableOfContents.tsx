'use client';

import { useState, useEffect } from 'react';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { cn } from '@/lib/utils';

interface TOCItem {
  id: string;
  title: string;
  level: number;
}

interface TableOfContentsProps {
  items: TOCItem[];
}

export function TableOfContents({ items }: TableOfContentsProps) {
  const [activeId, setActiveId] = useState<string>('');

  useEffect(() => {
    const observer = new IntersectionObserver(
      entries => {
        entries.forEach(entry => {
          if (entry.isIntersecting) {
            setActiveId(entry.target.id);
          }
        });
      },
      { rootMargin: '-20% 0% -35% 0%' }
    );

    items.forEach(item => {
      const element = document.getElementById(item.id);
      if (element) observer.observe(element);
    });

    return () => observer.disconnect();
  }, [items]);

  if (items.length === 0) return null;

  const scrollToSection = (id: string) => {
    const element = document.getElementById(id);
    if (element) {
      element.scrollIntoView({ behavior: 'smooth', block: 'start' });
    }
  };

  return (
    <Card className="sticky top-8">
      <CardHeader className="pb-3">
        <CardTitle className="text-base">Table of Contents</CardTitle>
      </CardHeader>
      <CardContent className="pt-0">
        <nav className="space-y-1">
          {items.map(item => (
            <button
              key={item.id}
              onClick={() => scrollToSection(item.id)}
              className={cn(
                'block w-full text-left text-sm py-1 px-2 rounded transition-colors hover:bg-accent hover:text-accent-foreground',
                item.level === 3 && 'ml-4',
                item.level === 4 && 'ml-8',
                activeId === item.id &&
                  'bg-accent text-accent-foreground font-medium'
              )}
            >
              {item.title}
            </button>
          ))}
        </nav>
      </CardContent>
    </Card>
  );
}
