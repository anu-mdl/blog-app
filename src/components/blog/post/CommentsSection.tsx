'use client';

import type React from 'react';

import { useState } from 'react';
import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar';
import { Button } from '@/components/ui/button';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { Textarea } from '@/components/ui/textarea';
import { Heart, MessageCircle, MoreHorizontal } from 'lucide-react';
import { formatDate } from '@/lib/utils';
import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
import { CommentsRecordExtended } from '@/api/extended_types';
import { pb } from '@/lib/pb';
import { useParams } from 'next/navigation';
import { Skeleton } from '@/components/ui/skeleton';
import Cookies from 'js-cookie';
import { pocketbase } from '@/api/pocketbase';
import { pocketbaseClient } from '@/api/pocketbase-client';
import { CommentComponent } from './CommentComponent';

const CommentSkeleton = () => (
  <div className="flex gap-3">
    <Skeleton className="h-8 w-8 rounded-full" />
    <div className="flex-1 space-y-2">
      <Skeleton className="h-4 w-1/4" />
      <Skeleton className="h-4 w-full" />
    </div>
  </div>
);

export function CommentsSection() {
  const params = useParams<{ id: string }>();
  const [isSubmitting, setIsSubmitting] = useState(false);
  const [newComment, setNewComment] = useState('');
  let authorData = pb.authStore.model;

  const { data, error, isLoading } = useQuery({
    queryKey: ['comments'],
    queryFn: async () => {
      const response = await pb
        .collection('comments')
        .getFullList<CommentsRecordExtended>({
          filter: `post="${params.id}"`,
          sort: 'created',
          expand: 'author'
        });
      return response;
    }
  });

  const usePostComment = () => {
    const queryClient = useQueryClient();

    return useMutation({
      mutationFn: async ({
        content,
        postId,
        authorId
      }: {
        content: string;
        postId: string;
        authorId: string;
      }) => {
        const response = await pb.collection('comments').create({
          content,
          post: postId,
          author: authorId,
          likes: 0,
          replies: []
        });
        return response;
      },
      onSuccess: () => {
        queryClient.invalidateQueries({ queryKey: ['comments'] });
      },
      onError: (error: Error) => {
        console.error('Failed to post comment:', error);
      }
    });
  };

  const postCommentMutation = usePostComment();

  const handleSubmitComment = async (e: React.FormEvent) => {
    e.preventDefault();
    const pb = await pocketbaseClient();

    if (!newComment.trim()) return;

    setIsSubmitting(true);

    const authorId = pb.authStore.model?.id;
    if (!authorId) {
      try {
        const authData = await pb.collection('author').authRefresh();
        const refreshedAuthorauthorId = pb.authStore.model?.id;

        if (!refreshedAuthorauthorId) {
          return <div>Unable to retrieve author ID after refresh</div>;
        }

        authorData = await pb
          .collection('author')
          .getOne(refreshedAuthorauthorId);
      } catch (refreshError) {
        console.error('Error refreshing auth:', refreshError);
        return;
      }
    }

    if (authorId === undefined) {
      return;
    }

    try {
      await postCommentMutation.mutateAsync({
        content: newComment.trim(),
        postId: params.id,
        authorId
      });

      setNewComment('');
    } catch (error) {
      console.error('Error posting comment:', error);
    } finally {
      setIsSubmitting(false);
    }
  };

  return (
    <Card className="min-w-0 overflow-hidden">
      <CardHeader>
        <CardTitle className="text-lg">
          Comments ({data?.length || 0})
        </CardTitle>
      </CardHeader>
      <CardContent className="space-y-6 min-w-0">
        <form onSubmit={handleSubmitComment} className="space-y-4">
          <Textarea
            placeholder="Share your thoughts..."
            value={newComment}
            onChange={e => setNewComment(e.target.value)}
            className="min-h-[100px] resize-none"
          />
          <div className="flex justify-end">
            <Button type="submit" disabled={!newComment.trim() || isSubmitting}>
              {isSubmitting ? 'Posting...' : 'Post Comment'}
            </Button>
          </div>
        </form>

        <div className="space-y-6 min-w-0">
          {isLoading && (
            <>
              {Array.from({ length: 3 }).map((_, i) => (
                <CommentSkeleton key={i} />
              ))}
            </>
          )}

          {error && (
            <div className="text-sm text-destructive">
              Failed to load comments. Please try again.
            </div>
          )}
          {data?.map(comment => (
            <CommentComponent key={comment.id} comment={comment} />
          ))}
        </div>
      </CardContent>
    </Card>
  );
}
