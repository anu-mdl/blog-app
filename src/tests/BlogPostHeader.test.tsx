import { BlogPostHeader } from '@/components/blog/post/BlogPostHeader';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render, screen } from '@testing-library/react';

jest.mock('@/lib/pb', () => ({
  pb: {
    authStore: {
      model: null
    },
    collection: jest.fn().mockReturnThis(),
    getList: jest.fn(),
    getFirstListItem: jest.fn(),
    create: jest.fn(),
    delete: jest.fn(),
    files: {
      getURL: jest
        .fn()
        .mockImplementation((_, filename) =>
          filename ? `https://mocked.pb.url/files/${filename}` : '/image.svg'
        )
    }
  }
}));

jest.mock('lucide-react', () => ({
  Calendar: () => <span>CalendarIcon</span>,
  Heart: () => <span>HeartIcon</span>,
  MessageSquare: () => <span>MessageSquareIcon</span>
}));

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => <img {...props} />
}));

jest.mock('@/components/ui/avatar', () => ({
  Avatar: jest.fn(({ children }) => <div>{children}</div>),
  AvatarFallback: jest.fn(({ children }) => <div>{children}</div>),
  AvatarImage: jest.fn(({ src, alt }) => <img src={src} alt={alt} />)
}));

jest.mock('@/components/ui/badge', () => ({
  Badge: jest.fn(({ children }) => <span>{children}</span>)
}));

jest.mock('@/components/ui/button', () => ({
  Button: jest.fn(({ children, ...props }) => (
    <button {...props}>{children}</button>
  ))
}));

describe('BlogPostHeader', () => {
  const mockPost = {
    id: 'post123',
    category: 'Technology',
    title: 'Test Post Title',
    excerpt: 'This is a test post excerpt',
    created: '2023-01-01T12:00:00.000Z',
    tags: 'react,nextjs,testing',
    image: 'test-image.jpg',
    expand: {
      author: {
        id: 'user456',
        username: 'testuser',
        avatar: 'avatar.jpg'
      }
    }
  };

  const queryClient = new QueryClient();

  const renderComponent = () => {
    return render(
      <QueryClientProvider client={queryClient}>
        <BlogPostHeader post={mockPost as any} />
      </QueryClientProvider>
    );
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with authenticated user', async () => {
    require('@/lib/pb').pb.authStore.model = { id: 'user123' };

    require('@/lib/pb')
      .pb.collection('likes')
      .getList.mockResolvedValueOnce({ totalItems: 10 });
    require('@/lib/pb')
      .pb.collection('likes')
      .getFirstListItem.mockResolvedValueOnce({
        id: 'like123'
      });

    queryClient.setQueryData(['comments-count', mockPost.id], 5);

    const { asFragment } = renderComponent();

    expect(await screen.findByText('10')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly without authenticated user', async () => {
    require('@/lib/pb').pb.authStore.model = null;

    require('@/lib/pb')
      .pb.collection('likes')
      .getList.mockResolvedValueOnce({ totalItems: 7 });
    require('@/lib/pb')
      .pb.collection('likes')
      .getFirstListItem.mockRejectedValueOnce(new Error('Not found'));

    queryClient.setQueryData(['comments-count', mockPost.id], 3);

    const { asFragment } = renderComponent();

    expect(await screen.findByText('7')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with no author avatar', async () => {
    const postWithoutAvatar = {
      ...mockPost,
      expand: {
        author: {
          id: 'user456',
          username: 'testuser',
          avatar: null
        }
      }
    };

    require('@/lib/pb').pb.authStore.model = { id: 'user123' };

    require('@/lib/pb')
      .pb.collection('likes')
      .getList.mockResolvedValueOnce({ totalItems: 10 });
    require('@/lib/pb')
      .pb.collection('likes')
      .getFirstListItem.mockResolvedValueOnce({
        id: 'like123'
      });

    const { asFragment } = render(
      <QueryClientProvider client={queryClient}>
        <BlogPostHeader post={postWithoutAvatar as any} />
      </QueryClientProvider>
    );

    expect(await screen.findByText('10')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with no image', async () => {
    const postWithoutImage = {
      ...mockPost,
      image: null
    };

    require('@/lib/pb').pb.authStore.model = { id: 'user123' };

    require('@/lib/pb')
      .pb.collection('likes')
      .getList.mockResolvedValueOnce({ totalItems: 10 });
    require('@/lib/pb')
      .pb.collection('likes')
      .getFirstListItem.mockResolvedValueOnce({
        id: 'like123'
      });

    const { asFragment } = render(
      <QueryClientProvider client={queryClient}>
        <BlogPostHeader post={postWithoutImage as any} />
      </QueryClientProvider>
    );

    expect(await screen.findByText('10')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly while loading', () => {
    require('@/lib/pb').pb.authStore.model = { id: 'user123' };

    const { asFragment } = renderComponent();

    expect(asFragment()).toMatchSnapshot();
  });
});
