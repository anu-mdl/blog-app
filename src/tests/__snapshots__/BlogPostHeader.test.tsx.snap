// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostHeader renders correctly while loading 1`] = `
<DocumentFragment>
  <header
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <span>
        Technology
      </span>
      <h1
        class="text-3xl md:text-4xl lg:text-5xl font-bold tracking-tight leading-tight"
      >
        Test Post Title
      </h1>
      <p
        class="text-xl text-muted-foreground leading-relaxed"
      >
        This is a test post excerpt
      </p>
    </div>
    <div
      class="flex flex-col sm:flex-row sm:items-center gap-4 text-sm text-muted-foreground"
    >
      <div
        class="flex items-center gap-3"
      >
        <div>
          <img
            alt="testuser"
            src="https://mocked.pb.url/files/avatar.jpg"
          />
          <div>
            t
          </div>
        </div>
        <div>
          <div
            class="font-medium mutationtext-foreground"
          >
            testuser
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-4"
      >
        <div
          class="flex items-center gap-1"
        >
          <span>
            CalendarIcon
          </span>
          <span>
            January 1, 2023
          </span>
        </div>
        <div
          class="flex items-center gap-1"
        >
          <span>
            MessageSquareIcon
          </span>
          <span>
            3
          </span>
        </div>
      </div>
    </div>
    <div
      class="relative aspect-video overflow-hidden rounded-lg"
    >
      <img
        alt="Cover"
        class="w-full h-full object-cover"
        height="600"
        src="https://mocked.pb.url/files/test-image.jpg"
        width="1200"
      />
    </div>
    <div
      class="flex flex-wrap gap-2 justify-between"
    >
      <div
        class="flex items-center gap-2"
      >
        <span>
          #react
        </span>
        <span>
          #nextjs
        </span>
        <span>
          #testing
        </span>
      </div>
      <button
        class="h-8 px-2 gap-1 text-xs transition-colors text-muted-foreground hover:text-red-600"
        variant="ghost"
      >
        <span>
          HeartIcon
        </span>
        <span>
          0
        </span>
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`BlogPostHeader renders correctly with authenticated user 1`] = `
<DocumentFragment>
  <header
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <span>
        Technology
      </span>
      <h1
        class="text-3xl md:text-4xl lg:text-5xl font-bold tracking-tight leading-tight"
      >
        Test Post Title
      </h1>
      <p
        class="text-xl text-muted-foreground leading-relaxed"
      >
        This is a test post excerpt
      </p>
    </div>
    <div
      class="flex flex-col sm:flex-row sm:items-center gap-4 text-sm text-muted-foreground"
    >
      <div
        class="flex items-center gap-3"
      >
        <div>
          <img
            alt="testuser"
            src="https://mocked.pb.url/files/avatar.jpg"
          />
          <div>
            t
          </div>
        </div>
        <div>
          <div
            class="font-medium mutationtext-foreground"
          >
            testuser
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-4"
      >
        <div
          class="flex items-center gap-1"
        >
          <span>
            CalendarIcon
          </span>
          <span>
            January 1, 2023
          </span>
        </div>
        <div
          class="flex items-center gap-1"
        >
          <span>
            MessageSquareIcon
          </span>
          <span>
            5
          </span>
        </div>
      </div>
    </div>
    <div
      class="relative aspect-video overflow-hidden rounded-lg"
    >
      <img
        alt="Cover"
        class="w-full h-full object-cover"
        height="600"
        src="https://mocked.pb.url/files/test-image.jpg"
        width="1200"
      />
    </div>
    <div
      class="flex flex-wrap gap-2 justify-between"
    >
      <div
        class="flex items-center gap-2"
      >
        <span>
          #react
        </span>
        <span>
          #nextjs
        </span>
        <span>
          #testing
        </span>
      </div>
      <button
        class="h-8 px-2 gap-1 text-xs transition-colors text-red-600 hover:text-red-700"
        variant="ghost"
      >
        <span>
          HeartIcon
        </span>
        <span>
          10
        </span>
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`BlogPostHeader renders correctly with no author avatar 1`] = `
<DocumentFragment>
  <header
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <span>
        Technology
      </span>
      <h1
        class="text-3xl md:text-4xl lg:text-5xl font-bold tracking-tight leading-tight"
      >
        Test Post Title
      </h1>
      <p
        class="text-xl text-muted-foreground leading-relaxed"
      >
        This is a test post excerpt
      </p>
    </div>
    <div
      class="flex flex-col sm:flex-row sm:items-center gap-4 text-sm text-muted-foreground"
    >
      <div
        class="flex items-center gap-3"
      >
        <div>
          <img
            alt="testuser"
            src="/image.svg"
          />
          <div>
            t
          </div>
        </div>
        <div>
          <div
            class="font-medium mutationtext-foreground"
          >
            testuser
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-4"
      >
        <div
          class="flex items-center gap-1"
        >
          <span>
            CalendarIcon
          </span>
          <span>
            January 1, 2023
          </span>
        </div>
        <div
          class="flex items-center gap-1"
        >
          <span>
            MessageSquareIcon
          </span>
          <span>
            3
          </span>
        </div>
      </div>
    </div>
    <div
      class="relative aspect-video overflow-hidden rounded-lg"
    >
      <img
        alt="Cover"
        class="w-full h-full object-cover"
        height="600"
        src="https://mocked.pb.url/files/test-image.jpg"
        width="1200"
      />
    </div>
    <div
      class="flex flex-wrap gap-2 justify-between"
    >
      <div
        class="flex items-center gap-2"
      >
        <span>
          #react
        </span>
        <span>
          #nextjs
        </span>
        <span>
          #testing
        </span>
      </div>
      <button
        class="h-8 px-2 gap-1 text-xs transition-colors text-muted-foreground hover:text-red-600"
        variant="ghost"
      >
        <span>
          HeartIcon
        </span>
        <span>
          10
        </span>
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`BlogPostHeader renders correctly with no image 1`] = `
<DocumentFragment>
  <header
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <span>
        Technology
      </span>
      <h1
        class="text-3xl md:text-4xl lg:text-5xl font-bold tracking-tight leading-tight"
      >
        Test Post Title
      </h1>
      <p
        class="text-xl text-muted-foreground leading-relaxed"
      >
        This is a test post excerpt
      </p>
    </div>
    <div
      class="flex flex-col sm:flex-row sm:items-center gap-4 text-sm text-muted-foreground"
    >
      <div
        class="flex items-center gap-3"
      >
        <div>
          <img
            alt="testuser"
            src="https://mocked.pb.url/files/avatar.jpg"
          />
          <div>
            t
          </div>
        </div>
        <div>
          <div
            class="font-medium mutationtext-foreground"
          >
            testuser
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-4"
      >
        <div
          class="flex items-center gap-1"
        >
          <span>
            CalendarIcon
          </span>
          <span>
            January 1, 2023
          </span>
        </div>
        <div
          class="flex items-center gap-1"
        >
          <span>
            MessageSquareIcon
          </span>
          <span>
            3
          </span>
        </div>
      </div>
    </div>
    <div
      class="relative aspect-video overflow-hidden rounded-lg"
    >
      <img
        alt="Cover"
        class="w-full h-full object-cover"
        height="600"
        src="/image.svg"
        width="1200"
      />
    </div>
    <div
      class="flex flex-wrap gap-2 justify-between"
    >
      <div
        class="flex items-center gap-2"
      >
        <span>
          #react
        </span>
        <span>
          #nextjs
        </span>
        <span>
          #testing
        </span>
      </div>
      <button
        class="h-8 px-2 gap-1 text-xs transition-colors text-red-600 hover:text-red-700"
        variant="ghost"
      >
        <span>
          HeartIcon
        </span>
        <span>
          10
        </span>
      </button>
    </div>
  </header>
</DocumentFragment>
`;

exports[`BlogPostHeader renders correctly without authenticated user 1`] = `
<DocumentFragment>
  <header
    class="space-y-6"
  >
    <div
      class="space-y-4"
    >
      <span>
        Technology
      </span>
      <h1
        class="text-3xl md:text-4xl lg:text-5xl font-bold tracking-tight leading-tight"
      >
        Test Post Title
      </h1>
      <p
        class="text-xl text-muted-foreground leading-relaxed"
      >
        This is a test post excerpt
      </p>
    </div>
    <div
      class="flex flex-col sm:flex-row sm:items-center gap-4 text-sm text-muted-foreground"
    >
      <div
        class="flex items-center gap-3"
      >
        <div>
          <img
            alt="testuser"
            src="https://mocked.pb.url/files/avatar.jpg"
          />
          <div>
            t
          </div>
        </div>
        <div>
          <div
            class="font-medium mutationtext-foreground"
          >
            testuser
          </div>
        </div>
      </div>
      <div
        class="flex items-center gap-4"
      >
        <div
          class="flex items-center gap-1"
        >
          <span>
            CalendarIcon
          </span>
          <span>
            January 1, 2023
          </span>
        </div>
        <div
          class="flex items-center gap-1"
        >
          <span>
            MessageSquareIcon
          </span>
          <span>
            3
          </span>
        </div>
      </div>
    </div>
    <div
      class="relative aspect-video overflow-hidden rounded-lg"
    >
      <img
        alt="Cover"
        class="w-full h-full object-cover"
        height="600"
        src="https://mocked.pb.url/files/test-image.jpg"
        width="1200"
      />
    </div>
    <div
      class="flex flex-wrap gap-2 justify-between"
    >
      <div
        class="flex items-center gap-2"
      >
        <span>
          #react
        </span>
        <span>
          #nextjs
        </span>
        <span>
          #testing
        </span>
      </div>
      <button
        class="h-8 px-2 gap-1 text-xs transition-colors text-muted-foreground hover:text-red-600"
        disabled=""
        variant="ghost"
      >
        <span>
          HeartIcon
        </span>
        <span>
          7
        </span>
      </button>
    </div>
  </header>
</DocumentFragment>
`;
